src/third_party/field_5x52_asm_impl_cryptopt.h

```diff
@@ -0,0 +1,343 @@
+/**
+ * Generated by CryptOpt (https://github.com/0xADE1A1DE/CryptOpt)
+ */
+
+#ifndef SECP256K1_FIELD_INNER5X52_IMPL_H
+#define SECP256K1_FIELD_INNER5X52_IMPL_H
+
+#include "../util.h"
+
+SECP256K1_INLINE static void
+secp256k1_fe_mul_inner(uint64_t *r, const uint64_t *a,
+                       const uint64_t *SECP256K1_RESTRICT b) {
```

Comment:
- This can go on one line. (We don't have a fixed value for max length of lines, but they tend to be longer than shorter.)
 - I know you took the `SECP256K1_RESTRICT` from the old code, but I think it can be dropped. It's rather meaningless here. (What is the compiler supposed to optimize here?! It's just asm.) Or am I overlooking something?

Reply:
> * (What is the compiler supposed to optimize here?! It's just asm.)

Now that I think about it again, I suggest leaving it in. It's there now, and whether to remove it is probably a separate discussion.

But in general, can you confirm that the code here is intended to stick to the rules laid out in `field.h`?
```
 /* (...)
 * r and a may point to the same object, but neither can be equal to b. (...)
 */
static void secp256k1_fe_mul(secp256k1_fe *r, const secp256k1_fe *a, const secp256k1_fe *b); 
```

Reply:
Neither the Optimizer, nor the checker enforces that property. But manually checking it at the moment shows that it is the case. Maybe it's possible to check that on the fiat side. I'm tracking that in issue https://github.com/0xADE1A1DE/CryptOpt/issues/167 for the cryptopt side.